compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1074,6 +1074,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
   }
 
   def typedQuote(tree: untpd.Quote, pt: Type)(implicit ctx: Context): Tree = track("typedQuote") {
+    ctx.compilationUnit.containsQuotesOrSplices = true
```

Comment:
Are you sure this is needed? The field should be set in PostTyper anyway.

Reply:
`tests/run/quote-and-splice` fails otherwise. I will move it to PostTyper.

Reply:
Actually it is not needed anymore. I forgot you added the logic for that in `handleMeta` after my comment on #3634.